### ISSUE TITLE
calcite-block: add setFocus() method

### ISSUE THREAD

**Issue by AdelheidF** [Esri/calcite-components#6944]
### Description

I have a use case where the tab order is supposed to go to a floating action-pad before it comes back to the flow-item and the block. But currently I cannot set the focus onto the block programmatically.

![image](https://github.com/Esri/calcite-components/assets/44481292/8e3788cf-4973-4608-b190-d842f65e1265)
![image](https://github.com/Esri/calcite-components/assets/44481292/ec30ff83-1580-4cb9-a9ef-dd5287122eac)

cc @jcfranco 

### Acceptance Criteria

new setFocus method

### Which Component

calcite-block

### Priority impact

p1 - need for current milestone

### Esri team

ArcGIS Map Viewer

**Comment by geospatialem** [comment]
Verified in `1.5.0-next.8`

![block-set-focus-verify](https://github.com/Esri/calcite-components/assets/5023024/445c576a-15e2-4feb-a45a-1fe1a6901b84)
